lib/ash/resource/transformers/create_join_relationship.ex

```diff
@@ -26,7 +26,8 @@ defmodule Ash.Resource.Transformers.CreateJoinRelationship do
           name: relationship.join_relationship,
           destination: relationship.through,
           destination_field: relationship.source_field_on_join_table,
-          source_field: relationship.source_field
+          source_field: relationship.source_field,
+          private?: relationship.private?
```

Comment:
This should actually just be `private?: true`. If they want the join relationship to be public they can declare the join relationship themselves and point the `many_to_many` at that join relationship that they created.